app/models/room.rb

```diff
@@ -9,4 +9,14 @@ class Room < ApplicationRecord
   belongs_to :current_record, foreign_key: :current_record_id, class_name: "RoomPlaylistRecord", optional: true
   has_one :current_song, through: :current_record, source: :song
   belongs_to :team
+
+  def idle!
+    just_finished = current_record.present?
+    update!(current_record: nil, playing_until: nil, waiting_songs: false)
+
+    return unless just_finished
+
+    BroadcastNowPlayingWorker.perform_async(self.id)
```

Comment:
I wasn't 100% happy bout leaving these worker calls in here. It may be argued that this refactor is not necessary.

Reply:
You know, I think that the early return around this `just_finished` concept was important to us when the (at the time) `QueueManagemenWorker` was responsible for queuing itself over and over.  The expectation there was that if no one was playing music in a room, it'd hit this "empty" block every one or two seconds, so we wanted to skip enqueuing these broadcasts.

With the event loop thing, we're more or less guaranteed that `just_finished` should always be true.  I like encapsulating the `update!` in this `idle!` method -- perhaps we could remove `just_finished` and pull the broadcasts back out and invoke them just after `idle!`?

Reply:
Oh we early return `idle!` so it's not quite that straightforward, but still perhaps something to think about.